packages/platforms/accton/x86-64/as7535-28xb/onlp/builds/x86_64_accton_as7535_28xb/module/src/sfpi.c

```diff
@@ -308,6 +318,26 @@ onlp_sfpi_control_set(int port, onlp_sfp_control_t control, int value)
             return ONLP_STATUS_OK;
         }
     }
+    case ONLP_SFP_CONTROL_SOFT_RATE_SELECT: {
+        VALIDATE_SFP(port);
+
+        present = onlp_sfpi_is_present(port);
+        if (present == 1) {
+            mbit_identifier = onlp_sfpi_dev_readb(port, MBIT_IDENTIFIER_ADDR, MBIT_OFFSET);
+            if(mbit_identifier == RATE_SELECT) {
+                onlp_sfpi_dev_writeb(port, PORT_EEPROM_DEVADDR, MULTIRATE_OFFSET, value);
```

Comment:
The Soft Rate_Select is controlled by bit 3 of the Optional Status/Control register 0x6E.
It is better to read out the current value of register 0x6E and then overwrite only bit 3 with the parameter 'value' to prevent mis-writing to the other bits.

Reply:
addressed